instrumentor/runtimemigration/runtimemigration.go

```diff
@@ -137,13 +138,20 @@ func (m *MigrationRunnable) fetchAndProcessDeployments(ctx context.Context, kube
 
 					err := handleContainerRuntimeDetailsUpdate(
 						containerObject,
-						*originalWorkloadEnvVar,
+						originalWorkloadEnvVar,
 						runtimeDetailsByContainer,
 					)
 					if err != nil {
 						return fmt.Errorf("failed to process container %s in deployment %s: %v", containerObject.Name, dep.Name, err)
 					}
 				}
+
+				controllerutil.CreateOrPatch(ctx, kubeClient, &dep, func() error {
+					originalWorkloadEnvVar.SetModifiedSinceCreated()
```

Comment:
we are going to execute this for every deployment, and everytime a new instrumentor starts, regardless if anything actually changed or not.

While it should be a no-op, I wonder if it can introduce any issues or race conditions, and if we can run it only when actually necessary